<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Peer Assignment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a Db2 database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the links below to download and save the datasets (.CSV files):

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Census Data</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Public Schools</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Crime Data</a>

**NOTE:** Ensure you have downloaded the datasets using the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**

Let us now load the ipython-sql  extension and establish a connection with the database

* Here you will be loading the csv files into the pandas Dataframe and then loading the data into the above mentioned sqlite tables.

* Next you will be connecting to the sqlite database  **FinalDB**.

Refer to the previous lab for hints .

<a href ="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Module%205/DB0201EN-Week3-1-4-Analyzing_SQLite.ipynb">Hands-on Lab: Analyzing a real World Data Set</a>





In [2]:
%load_ext sql

In [3]:
import csv, sqlite3

con = sqlite3.connect("FinalDB.db")
cur = con.cursor()

In [4]:
!pip install -q pandas==1.1.5

In [5]:
%sql sqlite:///FinalDB.db



'Connected: @FinalDB.db'

## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [6]:
import pandas as pd

In [7]:
crime_data = pd.read_csv('ChicagoCrimeData.csv')
%sql PERSIST crime_data

 * sqlite:///FinalDB.db


ValueError: Table 'crime_data' already exists.

In [8]:
%sql select count (*) as total_num_rows from crime_data

 * sqlite:///FinalDB.db
Done.


[(533,)]

### Problem 2

##### List community areas with per capita income less than 11000.


In [9]:
census_data = pd.read_csv('ChicagoCensusData.csv')
%sql PERSIST census_data

 * sqlite:///FinalDB.db


ValueError: Table 'census_data' already exists.

In [35]:
%sql select COMMUNITY_AREA_NAME, PER_CAPITA_INCOME from census_data where PER_CAPITA_INCOME < 11000

 * sqlite:///FinalDB.db
Done.


[('West Garfield Park', 10934),
 ('South Lawndale', 10402),
 ('Fuller Park', 10432),
 ('Riverdale', 8201)]

In [11]:
%sql select * from census_data

 * sqlite:///FinalDB.db
Done.


[(0, 1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0),
 (1, 2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0),
 (2, 3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0),
 (3, 4.0, 'Lincoln Square', 3.4, 10.9, 8.2, 13.4, 25.5, 37524, 17.0),
 (4, 5.0, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0),
 (5, 6.0, 'Lake View', 1.1, 11.4, 4.7, 2.6, 17.0, 60058, 5.0),
 (6, 7.0, 'Lincoln Park', 0.8, 12.3, 5.1, 3.6, 21.5, 71551, 2.0),
 (7, 8.0, 'Near North Side', 1.9, 12.9, 7.0, 2.5, 22.6, 88669, 1.0),
 (8, 9.0, 'Edison Park', 1.1, 3.3, 6.5, 7.4, 35.3, 40959, 8.0),
 (9, 10.0, 'Norwood Park', 2.0, 5.4, 9.0, 11.5, 39.5, 32875, 21.0),
 (10, 11.0, 'Jefferson Park', 2.7, 8.6, 12.4, 13.4, 35.5, 27751, 25.0),
 (11, 12.0, 'Forest Glen', 1.1, 7.5, 6.8, 4.9, 40.5, 44164, 11.0),
 (12, 13.0, 'North Park', 3.9, 13.2, 9.9, 14.4, 39.0, 26576, 33.0),
 (13, 14.0, 'Albany Park', 11.3, 19.2, 10.0, 32.9, 32.0, 21323, 53.0),
 (14, 15.0, 'Portage Park', 4.1, 11.6, 12.6, 19.3, 34.0, 24336, 35.0),
 (15, 16.0, 'Irving Park', 6.3, 13.1, 10.0, 22.4, 31.6, 27249, 34.0),
 (16, 17.0, 'Dunning', 5.2, 10.6, 10.0, 16.2, 33.6, 26282, 28.0),
 (17, 18.0, 'Montclaire', 8.1, 15.3, 13.8, 23.5, 38.6, 22014, 50.0),
 (18, 19.0, 'Belmont Cragin', 10.8, 18.7, 14.6, 37.3, 37.3, 15461, 70.0),
 (19, 20.0, 'Hermosa', 6.9, 20.5, 13.1, 41.6, 36.4, 15089, 71.0),
 (20, 21.0, 'Avondale', 6.0, 15.3, 9.2, 24.7, 31.0, 20039, 42.0),
 (21, 22.0, 'Logan Square', 3.2, 16.8, 8.2, 14.8, 26.2, 31908, 23.0),
 (22, 23.0, 'Humboldt park', 14.8, 33.9, 17.3, 35.4, 38.0, 13781, 85.0),
 (23, 24.0, 'West Town', 2.3, 14.7, 6.6, 12.9, 21.7, 43198, 10.0),
 (24, 25.0, 'Austin', 6.3, 28.6, 22.6, 24.4, 37.9, 15957, 73.0),
 (25, 26.0, 'West Garfield Park', 9.4, 41.7, 25.8, 24.5, 43.6, 10934, 92.0),
 (26, 27.0, 'East Garfield Park', 8.2, 42.4, 19.6, 21.3, 43.2, 12961, 83.0),
 (27, 28.0, 'Near West Side', 3.8, 20.6, 10.7, 9.6, 22.2, 44689, 15.0),
 (28, 29.0, 'North Lawndale', 7.4, 43.1, 21.2, 27.6, 42.7, 12034, 87.0),
 (29, 30.0, 'South Lawndale', 15.2, 30.7, 15.8, 54.8, 33.8, 10402, 96.0),
 (30, 31.0, 'Lower West Side', 9.6, 25.8, 15.8, 40.7, 32.6, 16444, 76.0),
 (31, 32.0, 'Loop', 1.5, 14.7, 5.7, 3.1, 13.5, 65526, 3.0),
 (32, 33.0, 'Near South Side', 1.3, 13.8, 4.9, 7.4, 21.8, 59077, 7.0),
 (33, 34.0, 'Armour Square', 5.7, 40.1, 16.7, 34.5, 38.3, 16148, 82.0),
 (34, 35.0, 'Douglas', 1.8, 29.6, 18.2, 14.3, 30.7, 23791, 47.0),
 (35, 36.0, 'Oakland', 1.3, 39.7, 28.7, 18.4, 40.4, 19252, 78.0),
 (36, 37.0, 'Fuller Park', 3.2, 51.2, 33.9, 26.6, 44.9, 10432, 97.0),
 (37, 38.0, 'Grand Boulevard', 3.3, 29.3, 24.3, 15.9, 39.5, 23472, 57.0),
 (38, 39.0, 'Kenwood', 2.4, 21.7, 15.7, 11.3, 35.4, 35911, 26.0),
 (39, 40.0, 'Washington Park', 5.6, 42.1, 28.6, 25.4, 42.8, 13785, 88.0),
 (40, 41.0, 'Hyde Park', 1.5, 18.4, 8.4, 4.3, 26.2, 39056, 14.0),
 (41, 42.0, 'Woodlawn', 2.9, 30.7, 23.4, 16.5, 36.1, 18672, 58.0),
 (42, 43.0, 'South Shore', 2.8, 31.1, 20.0, 14.0, 35.7, 19398, 55.0),
 (43, 44.0, 'Chatham', 3.3, 27.8, 24.0, 14.5, 40.3, 18881, 60.0),
 (44, 45.0, 'Avalon Park', 1.4, 17.2, 21.1, 10.6, 39.3, 24454, 41.0),
 (45, 46.0, 'South Chicago', 4.7, 29.8, 19.7, 26.6, 41.1, 16579, 75.0),
 (46, 47.0, 'Burnside', 6.8, 33.0, 18.6, 19.3, 42.7, 12515, 79.0),
 (47, 48.0, 'Calumet Heights', 2.1, 11.5, 20.0, 11.0, 44.0, 28887, 38.0),
 (48, 49.0, 'Roseland', 2.5, 19.8, 20.3, 16.9, 41.2, 17949, 52.0),
 (49, 50.0, 'Pullman', 1.5, 21.6, 22.8, 13.1, 38.6, 20588, 51.0),
 (50, 51.0, 'South Deering', 4.0, 29.2, 16.3, 21.0, 39.5, 14685, 65.0),
 (51, 52.0, 'East Side', 6.8, 19.2, 12.1, 31.9, 42.8, 17104, 64.0),
 (52, 53.0, 'West Pullman', 3.3, 25.9, 19.4, 20.5, 42.1, 16563, 62.0),
 (53, 54.0, 'Riverdale', 5.8, 56.5, 34.6, 27.5, 51.5, 8201, 98.0),
 (54, 55.0, 'Hegewisch', 3.3, 17.1, 9.6, 19.2, 42.9, 22677, 44.0),
 (55, 56.0, 'Garfield Ridge', 2.6, 8.8, 11.3, 19.3, 38.1, 26353, 32.0),
 (56, 57.0, 'Archer Heights', 8.5, 14.1, 16.5, 35.9, 39.2, 16134, 67.0),
 (57, 58.0, 'Brighton Park', 14.4, 23.6, 13.9, 45.1, 39.3, 13089

### Problem 3

##### List all case numbers for crimes  involving minors?(children are not considered minors for the purposes of crime analysis)


In [38]:
%%sql
SELECT ID, CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION FROM crime_data
WHERE DESCRIPTION LIKE '%MINOR%'

 * sqlite:///FinalDB.db
Done.


[(3987219, 'HL266884', 'LIQUOR LAW VIOLATION', 'SELL/GIVE/DEL LIQUOR TO MINOR'),
 (3266814, 'HK238408', 'LIQUOR LAW VIOLATION', 'ILLEGAL CONSUMPTION BY MINOR')]

### Problem 4

##### List all kidnapping crimes involving a child?


In [20]:
%%sql 
SELECT DISTINCT PRIMARY_TYPE as child_crime FROM crime_data
WHERE PRIMARY_TYPE LIKE '%CHILD%' 
UNION ALL
SELECT DISTINCT DESCRIPTION FROM  crime_data 
WHERE DESCRIPTION LIKE '%CHILD%'

 * sqlite:///FinalDB.db
Done.


[('OFFENSE INVOLVING CHILDREN',),
 ('AGG SEX ASSLT OF CHILD FAM MBR',),
 ('SEX ASSLT OF CHILD BY FAM MBR',),
 ('CHILD ABDUCTION/STRANGER',)]

### Problem 5

##### What kinds of crimes were recorded at schools?


In [26]:
public_school = pd.read_csv('ChicagoPublicSchools.csv')
%sql PERSIST public_school

 * sqlite:///FinalDB.db


'Persisted public_school'

In [39]:
%%sql
SELECT DISTINCT PRIMARY_TYPE, LOCATION_DESCRIPTION FROM crime_data
WHERE LOCATION_DESCRIPTION LIKE '%SCHOOL%' 
ORDER BY PRIMARY_TYPE

 * sqlite:///FinalDB.db
Done.


[('ASSAULT', 'SCHOOL, PUBLIC, GROUNDS'),
 ('BATTERY', 'SCHOOL, PUBLIC, GROUNDS'),
 ('BATTERY', 'SCHOOL, PUBLIC, BUILDING'),
 ('CRIMINAL DAMAGE', 'SCHOOL, PUBLIC, GROUNDS'),
 ('CRIMINAL TRESPASS', 'SCHOOL, PUBLIC, GROUNDS'),
 ('NARCOTICS', 'SCHOOL, PUBLIC, GROUNDS'),
 ('NARCOTICS', 'SCHOOL, PUBLIC, BUILDING'),
 ('PUBLIC PEACE VIOLATION', 'SCHOOL, PRIVATE, BUILDING'),
 ('PUBLIC PEACE VIOLATION', 'SCHOOL, PUBLIC, BUILDING')]

### Problem 6

##### List the average safety score for each type of school.


In [28]:
%%sql
ALTER TABLE public_school
RENAME COLUMN 'Elementary, Middle, or High School' to 'School_Type';

 * sqlite:///FinalDB.db
Done.


[]

In [29]:
%%sql 
SELECT school_type, AVG(safety_score) AS avg_safety_score
FROM public_school
GROUP BY school_type;

 * sqlite:///FinalDB.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [30]:
%%sql
SELECT COMMUNITY_AREA_NAME as area_Name, PERCENT_HOUSEHOLDS_BELOW_POVERTY as below_poverty FROM census_data
ORDER BY PERCENT_HOUSEHOLDS_BELOW_POVERTY DESC
LIMIT 5
;

 * sqlite:///FinalDB.db
Done.


[('Riverdale', 56.5),
 ('Fuller Park', 51.2),
 ('Englewood', 46.6),
 ('North Lawndale', 43.1),
 ('East Garfield Park', 42.4)]

### Problem 8

##### Which community area is most crime prone?


In [40]:
%%sql
SELECT census.COMMUNITY_AREA_NAME as AREA_NAME,  COUNT(crime.COMMUNITY_AREA_NUMBER) as num_of_crimes 
FROM crime_data as crime, census_data  as census
WHERE crime.COMMUNITY_AREA_NUMBER = census.COMMUNITY_AREA_NUMBER
GROUP BY crime.COMMUNITY_AREA_NUMBER
ORDER BY num_of_crimes DESC
LIMIT 1;

 * sqlite:///FinalDB.db
Done.


[('Austin', 43)]

Double-click **here** for a hint

<!--
Query for the 'community area number' that is most crime prone.
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [33]:
%%sql
select COMMUNITY_AREA_NAME, HARDSHIP_INDEX from census_data
where HARDSHIP_INDEX = (select max(HARDSHIP_INDEX) from census_data)


 * sqlite:///FinalDB.db
Done.


[('Riverdale', 98.0)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [34]:
%%sql 
SELECT DISTINCT census.COMMUNITY_AREA_NAME,crime.COMMUNITY_AREA_NUMBER  FROM census_data as census
JOIN crime_data as crime ON census.COMMUNITY_AREA_NUMBER =crime.COMMUNITY_AREA_NUMBER
WHERE crime.COMMUNITY_AREA_NUMBER = (SELECT COMMUNITY_AREA_NUMBER FROM crime_data
                                          GROUP BY COMMUNITY_AREA_NUMBER 
                                          ORDER BY COUNT(COMMUNITY_AREA_NUMBER)DESC
                                          LIMIT 1)

 * sqlite:///FinalDB.db
Done.


[('Austin', 25.0)]

Copyright © 2020 This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
